core/test/mpi/distributed/dense_communicator.cpp

```diff
@@ -0,0 +1,305 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <gtest/gtest.h>
+
+#include <ginkgo/core/distributed/dense_communicator.hpp>
+
+#include "core/test/utils/assertions.hpp"
+
+using gko::experimental::mpi::comm_index_type;
+
+class DenseCommunicator : public ::testing::Test {
+protected:
+    using part_type = gko::experimental::distributed::Partition<int, long>;
+    using map_type = gko::experimental::distributed::index_map<int, long>;
+
+    void SetUp() override { ASSERT_EQ(comm.size(), 6); }
+
+    gko::experimental::mpi::DenseCommunicator create_default_comm()
+    {
+        auto part = gko::share(part_type::build_from_global_size_uniform(
+            ref, comm.size(), comm.size() * 3));
+        gko::array<long> recv_connections[] = {{ref, {3, 5, 10, 11}},
+                                               {ref, {0, 1, 7, 12, 13}},
+                                               {ref, {3, 4, 17}},
+                                               {ref, {1, 2, 12, 14}},
+                                               {ref, {4, 5, 9, 10, 16, 15}},
+                                               {ref, {8, 12, 13, 14}}};
+        auto imap = map_type{ref, part, comm.rank(), recv_connections[rank]};
+
+
+        return {comm, imap};
+    }
+
+    std::shared_ptr<gko::Executor> ref = gko::ReferenceExecutor::create();
+    gko::experimental::mpi::communicator comm = MPI_COMM_WORLD;
+    int rank = comm.rank();
+};
+
+
+TEST_F(DenseCommunicator, CanDefaultConstruct)
+{
+    gko::experimental::mpi::DenseCommunicator nhcomm{comm};
```

Comment:
nit: its named `nhcomm` and `spcomm`, probably because of previous usage of neighborhood comm and sparse comm, but can now be updated to `dense_comm` or `dcomm`.